CONTRIBUTING.md

```diff
@@ -39,6 +41,51 @@ To develop the `docs/` site:
 
 2. Run `pnpm run docs` in Vite's root folder.
 
+### Docs Translation Contribution
+
+To add a new language to the Vite docs, see [`vite-docs-template`](https://github.com/tony19/vite-docs-template/blob/main/.github/CONTRIBUTING.md).
+
+## Notes on Dependencies
+
+Vite aims to be lightweight, and this includes being aware of the number of npm dependencies and their size.
+
+We use Rollup to pre-bundle most dependencies before publishing! Therefore, most dependencies, even those used in runtime source code, should be added under `devDependencies` by default. This also creates the following constraints that we need to be aware of in the codebase.
+
+### Usage of `require()`
+
+In some cases, we intentionally lazy-require some dependencies to improve start-up performance. However, note that we cannot use simple `require('somedep')` calls since these are ignored in ESM files, so the dependency won't be included in the bundle, and the actual dependency won't even be there when published since they are in `devDependencies`.
+
+Instead, use `(await import('somedep')).default`.
+
+### Think Before Adding a Dependency
+
+Most deps should be added to `devDependencies` even if they are needed at runtime. Some exceptions are:
+
+- Type packages. Example: `@types/*`.
+- Deps that cannot be properly bundled due to binary files. Example: `esbuild`.
+- Deps that ship their own types that are used in Vite's own public types. Example: `rollup`.
+
+Avoid deps with large transitive dependencies that result in bloated size compared to the functionality it provides. For example, `http-proxy` itself is around 380kB in size, but `http-proxy-middleware` pulls in a ton of dependencies that make it 3MB(!) when a minimal custom middleware on top of `http-proxy` only requires a couple of lines of code.
```

Comment:
Updated the size calculated by package-phobia.
https://packagephobia.com/result?p=http-proxy
https://packagephobia.com/result?p=http-proxy-middleware